src/multi_vector_simulator/D2_model_constraints.py

```diff
@@ -75,35 +75,25 @@ def add_constraints(local_energy_system, dict_values, dict_model):
     - D2.test_add_constraints_net_zero_energy_requirement_is_true()
     - D2.test_add_constraints_net_zero_energy_requirement_is_false()
     """
-    count_added_constraints = 0
 
-    if dict_values[CONSTRAINTS][MINIMAL_RENEWABLE_FACTOR][VALUE] > 0:
-        # Add minimal renewable factor constraint
-        local_energy_system = constraint_minimal_renewable_share(
-            local_energy_system, dict_values, dict_model
-        )
-        count_added_constraints += 1
+    constraint_functions = {
+        MINIMAL_RENEWABLE_FACTOR: constraint_minimal_renewable_share,
+        MAXIMUM_EMISSIONS: constraint_maximum_emissions,
+        MINIMAL_DEGREE_OF_AUTONOMY: constraint_minimal_degree_of_autonomy,
+        NET_ZERO_ENERGY: constraint_net_zero_energy,
+    }
 
-    if dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE] is not None:
-        # Add maximum emissions constraint
-        local_energy_system = constraint_maximum_emissions(
-            local_energy_system, dict_values
-        )
-        count_added_constraints += 1
```

Comment:
```
    if MAXIMUM_EMISSIONS in dict_values[CONSTRAINTS] and dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE] is not None:
            # Add maximum emissions constraint
            local_energy_system = constraint_maximum_emissions(
            local_energy_system, dict_values
            )
           count_added_constraints += 1
```

Reply:
This does not work if `MAXIMUM_EMISSIONS` is not in the dict (which is the reason I introduce this change for back compatibility :))

Reply:
Why does it not work? It does not work if `CONSTRAINTS not in dict_values`, but if the key `MAXIMUM_EMISSIONS` is not in `dict_values[CONSTRAINTS]` it would exit the if-statement, would it not? I mean, this should definitly work, right:
```
    if MAXIMUM_EMISSIONS in dict_values[CONSTRAINTS]:
        if dict_values[CONSTRAINTS][MAXIMUM_EMISSIONS][VALUE] is not None:
            # Add maximum emissions constraint
            local_energy_system = constraint_maximum_emissions(
            local_energy_system, dict_values
            )
           count_added_constraints += 1
```

Reply:
> I mean, this should definitly work, right:
This works yes, but the previous need to evaluate both A and B in `if A and B` before being able to decide what is the outcome of the logical operation `A and B`. When evaluating B an error would be raised if MAXIMUM_EMISSIONS is not a key of the dict

Reply:
We can fix it this way, now that we have 4 constraints it is equivalent. My solution is scalable without copy paste for every new constraints one would add in the future. The fact that the tests pass should be an indicator that my proposition did not break something, in theory